fe/src/main/java/org/apache/doris/rewrite/FEFunctions.java

```diff
@@ -96,6 +126,36 @@ public static DateLiteral dateSub(LiteralExpr date, LiteralExpr day) throws Anal
         return dateAdd(date, new IntLiteral(-(int) day.getLongValue()));
     }
 
+    @FEFunction(name = "years_sub", argTypes = { "DATETIME", "INT" }, returnType = "DATETIME")
```

Comment:
Does this work when partition column type is `DATE`

Reply:
Yes , I have test it